internal/oplog/oplog_apply.go

```diff
@@ -151,15 +152,29 @@ func (oa *Apply) Run() error {
 			return fmt.Errorf("cannot unmarshal oplog document: %s", err)
 		}
 
-		if oa.fcheck(dest["ts"].(bson.MongoTimestamp), oa.stopAtTs) {
-			return nil
+		if ts, ok := dest["ts"].(bson.MongoTimestamp); ok {
+			if oa.fcheck(ts, oa.stopAtTs) {
+				return nil
+			}
+		} else {
+			return fmt.Errorf("oplog document ts field is not bson.MongoTimestamp, it is %T", dest["ts"])
 		}
 
 		icmd, ok := dest["op"]
 		if !ok {
 			return fmt.Errorf("invalid oplog document. there is no command")
 		}
 
+		if atomic.LoadUint32(&oa.debug) != 0 {
+			fmt.Printf("%#v\n", dest)
+		}
+
+		if ns, ok := dest["ns"]; ok {
+			if skip(ns.(string)) {
```

Comment:
Just in case, are we sure `ns` is always a `string`?

Reply:
yes, it is the database name